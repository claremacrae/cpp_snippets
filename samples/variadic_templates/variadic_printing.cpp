// Code from https://youtu.be/CU3VYN6xGzM?t=178
// C++ Weekly - Ep 10 Variadic Expansion Wrap-Up
// In godbolt at https://godbolt.org/z/4jIxgF

#include <iostream>

template<typename T>
void print_impl(const T& t)
{
    std::cout << t << '\n';
}

template<typename ... T>
void print(const T& ... t)
{
    (void)std::initializer_list<int>{ (print_impl(t), 0)...};
}

int main()
{
    print("Hello", 42, "World");
}
