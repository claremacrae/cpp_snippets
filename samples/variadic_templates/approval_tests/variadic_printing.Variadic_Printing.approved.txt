Hello
42
World

